lib/flop/filter.ex

```diff
@@ -182,14 +183,32 @@ defmodule Flop.Filter do
     if is_binary(v) and String.trim_leading(v) == "", do: nil, else: v
   end
 
+  defp value_type(%FieldInfo{ecto_type: type}, op, repo) do
+    value_type(type, op, repo)
+  end
+
+  defp value_type(:binary_id, _op, nil = _repo), do: :binary_id
+
+  defp value_type(:binary_id, _op, repo) do
+    if Code.ensure_loaded?(repo) do
+      case repo.__adapter__().loaders(:binary_id, :binary_id) do
+        [type, _] -> type
+        [type] -> type
```

Comment:
1. The type is overridden below for the `empty`, `not_empty`, and `(i)like_and/or` operators. While (i)like is not supported on binary IDs, it is still possible to use the empty operators on a nullable `binary_id` column (`field :some_other_id, :binary_id`).
2. For the `:in` and `:not_in` operators, the type is wrapped into an `{:array, type}` tuple below. We need to do that for binary IDs as well, or else the validation for those operators will be incorrect.
3. It's also possible to define something like `field :some_ids, {:array, :binary_id}` . The inner type is unwrapped below. We need to determine the type module for binary IDs in that case as well.